pkg/podset/podset.go

```diff
@@ -118,6 +127,12 @@ func (podSetInfo *PodSetInfo) Merge(o PodSetInfo) error {
 			podSetInfo.Tolerations = append(podSetInfo.Tolerations, t)
 		}
 	}
+	// make sure we don't duplicate schedulingGates
+	for _, t := range o.SchedulingGates {
+		if slices.Index(podSetInfo.SchedulingGates, t) == -1 {
```

Comment:
Could you switch from this to `HasGate` lib?
https://github.com/kubernetes-sigs/kueue/blob/4c063056616b1486f699fdb4461903abc8fa0a79/pkg/util/pod/pod.go#L26